### PR TITLE
move equivalent block to poll

### PR DESCRIPTION
**Pull request by tikisailor**
Equivalent block will be queried during poll creation instead of strategy and poll schema was adjusted. As a side effect, old poll will not work (could be updated manually in db)

### CODE REVIEW

File: src/web3/token-vote/graphql/graphql.service.ts [hunk 1]
```diff
@@ -49,4 +52,28 @@ export class GraphqlService {
         return result;
     }
 
+    async getEquivalentBlock(blockHeight, chainId) {
```
Comment (wpmonty) [inline src/web3/token-vote/graphql/graphql.service.ts:55]: I wonder if we need safeguards for out of bounds. Ethereum block 1 existed before Polygon ever did, so in this case Polygon would be excluded from the strategy completely. 

Might need to log it for later.

File: src/poll/poll.q.consumer.service.ts [hunk 2]
```diff
@@ -66,14 +68,22 @@ export class PollCreateConsumer {
...
+            let mainnetBlock = strategy.block_height.find(BlockHeight => BlockHeight.chain_id === '1').block;
+
+            if (mainnetBlock <= 0) {
+                const block = await web3Utils.getEthersProvider(1).getBlockNumber();
+                mainnetBlock = block - mainnetBlock;
+                strategy.block_height.find(BlockHeight => BlockHeight.chain_id === '1').block = mainnetBlock;
             }
+            
+            // add just polygon for now.
+            const polygonBlock = await this.gqlService.getEquivalentBlock(mainnetBlock, '137');
```
Comment (wpmonty) [inline src/poll/poll.q.consumer.service.ts:81]: What if FE passes polygon blockheight? We could skip this check.

File: src/web3/strategy/strategies/controller.bdao.ts [hunk 3]
```diff
@@ -37,54 +37,24 @@ export class BankTokenWeightedStrategy extends StrategyBaseController implements
...
-        blockHeight: number | null,
+        blockHeights: BlockHeight[],
         evmService: EvmService,
         graphqlService: GraphqlService,
         logger: Logger,
         // tokenWhitelistService: TokenWhitelistMongoService,
     ) {
 
-        let equivalentBlock = null;
+        const equivalentBlock = blockHeights.find((blockHeight) => blockHeight.chain_id === '137').block;
```
Comment (wpmonty) [inline src/web3/strategy/strategies/controller.bdao.ts:47]: maybe need to rename to `polygonBlockHeight` as we plan to support other chains long term.

Maybe log as tech debt.

**Review by wpmonty — APPROVED** [review]
Small comments, but looks good

File: src/web3/token-vote/graphql/graphql.service.ts (hunk 1 above)
Comment (tikisailor) [reply src/web3/token-vote/graphql/graphql.service.ts:55]: There is a safeguard on line 71. I don't remeber where the number comes from though probably needs double checking. Also we're not handling that case properly. I'll try to fix that along with some retry logic on rate limit

File: src/poll/poll.q.consumer.service.ts (hunk 2 above)
Comment (tikisailor) [reply src/poll/poll.q.consumer.service.ts:81]: You mean just pass current block for now? Yes we could do that. Could also do that here until we implement block height on FE.

Comment (tikisailor) [reply src/poll/poll.q.consumer.service.ts]: fixed to return current block if request fails

**Pull request merged**